server/service_test.go

```diff
@@ -838,142 +727,182 @@ func TestEmptyTxRoot(t *testing.T) {
-		"../testdata/signed-blinded-beacon-block-capella.json",
+func blindBlockToBlockResponse[P Payload](signedBlock P) *builderApi.VersionedSubmitBlindedBlockResponse {
+	switch block := any(signedBlock).(type) {
+	case *eth2ApiV1Bellatrix.SignedBlindedBeaconBlock:
+		header := block.Message.Body.ExecutionPayloadHeader
+		return &builderApi.VersionedSubmitBlindedBlockResponse{
+			Version: spec.DataVersionBellatrix,
+			Bellatrix: &bellatrix.ExecutionPayload{
+				ParentHash:    header.ParentHash,
+				FeeRecipient:  header.FeeRecipient,
+				StateRoot:     header.StateRoot,
+				ReceiptsRoot:  header.ReceiptsRoot,
+				LogsBloom:     header.LogsBloom,
+				PrevRandao:    header.PrevRandao,
+				BlockNumber:   header.BlockNumber,
+				GasLimit:      header.GasLimit,
+				GasUsed:       header.GasUsed,
+				Timestamp:     header.Timestamp,
+				ExtraData:     header.ExtraData,
+				BaseFeePerGas: header.BaseFeePerGas,
+				BlockHash:     header.BlockHash,
+				Transactions:  make([]bellatrix.Transaction, 0),
+			},
+		}
+	case *eth2ApiV1Capella.SignedBlindedBeaconBlock:
+		header := block.Message.Body.ExecutionPayloadHeader
+		return &builderApi.VersionedSubmitBlindedBlockResponse{
+			Version: spec.DataVersionCapella,
+			Capella: &capella.ExecutionPayload{
+				ParentHash:    header.ParentHash,
+				FeeRecipient:  header.FeeRecipient,
+				StateRoot:     header.StateRoot,
+				ReceiptsRoot:  header.ReceiptsRoot,
+				LogsBloom:     header.LogsBloom,
+				PrevRandao:    header.PrevRandao,
+				BlockNumber:   header.BlockNumber,
+				GasLimit:      header.GasLimit,
+				GasUsed:       header.GasUsed,
+				Timestamp:     header.Timestamp,
+				ExtraData:     header.ExtraData,
+				BaseFeePerGas: header.BaseFeePerGas,
+				BlockHash:     header.BlockHash,
+				Transactions:  make([]bellatrix.Transaction, 0),
+				Withdrawals:   make([]*capella.Withdrawal, 0),
+			},
+		}
+	case *eth2ApiV1Deneb.SignedBlindedBeaconBlock:
+		header := block.Message.Body.ExecutionPayloadHeader
+		commitments := block.Message.Body.BlobKZGCommitments
+		return &builderApi.VersionedSubmitBlindedBlockResponse{
+			Version: spec.DataVersionDeneb,
+			Deneb:   denebHeader(header, commitments),
+		}
+	case *eth2ApiV1Electra.SignedBlindedBeaconBlock:
+		header := block.Message.Body.ExecutionPayloadHeader
+		commitments := block.Message.Body.BlobKZGCommitments
+		return &builderApi.VersionedSubmitBlindedBlockResponse{
+			Version: spec.DataVersionElectra,
+			Electra: denebHeader(header, commitments),
+		}
 	}
-	for _, fn := range testPayloadsFiles {
-		t.Run(fn, func(t *testing.T) {
-			jsonFile, err := os.Open(fn)
+	return nil
+}
+
+func denebHeader(header *deneb.ExecutionPayloadHeader, kzgCommitments []deneb.KZGCommitment) *builderApiDeneb.ExecutionPayloadAndBlobsBundle {
+	numBlobs := len(kzgCommitments)
+	commitments := make([]deneb.KZGCommitment, numBlobs)
+	copy(commitments, kzgCommitments)
+	proofs := make([]deneb.KZGProof, numBlobs)
+	blobs := make([]deneb.Blob, numBlobs)
```

Comment:
This feels wrong. Where are proofs and blobs populated?

Reply:
You're right, seems like it was borked before already

Reply:
Ah nvm I get it now, the proofs are not part of the blindedBlock, only the blindedBlockResponse, so we can't get them from the blindedBlock itself. Same with the withdrawals and transactions. blindedBlock is kindof a subset of blindedBlockResponse